subcommand/inject-connect/command_test.go

```diff
@@ -52,6 +52,66 @@ func TestRun_FlagValidation(t *testing.T) {
 			},
 			expErr: "request must be <= limit: -default-sidecar-proxy-cpu-request value of \"50m\" is greater than the -default-sidecar-proxy-cpu-limit value of \"25m\"",
 		},
+		{
+			flags:  []string{"-consul-k8s-image", "foo", "-init-container-cpu-limit=unparseable"},
+			expErr: "-init-container-cpu-limit is invalid",
+		},
+		{
+			flags:  []string{"-consul-k8s-image", "foo", "-init-container-cpu-request=unparseable"},
+			expErr: "-init-container-cpu-request is invalid",
+		},
+		{
+			flags:  []string{"-consul-k8s-image", "foo", "-init-container-memory-limit=unparseable"},
+			expErr: "-init-container-memory-limit is invalid",
+		},
+		{
+			flags:  []string{"-consul-k8s-image", "foo", "-init-container-memory-request=unparseable"},
+			expErr: "-init-container-memory-request is invalid",
+		},
+		{
+			flags: []string{"-consul-k8s-image", "foo",
+				"-init-container-memory-request=50Mi",
+				"-init-container-memory-limit=25Mi",
+			},
+			expErr: "request must be <= limit: -init-container-memory-request value of \"50Mi\" is greater than the -init-container-memory-limit value of \"25Mi\"",
+		},
+		{
+			flags: []string{"-consul-k8s-image", "foo",
+				"-init-container-cpu-request=50m",
+				"-init-container-cpu-limit=25m",
+			},
+			expErr: "request must be <= limit: -init-container-cpu-request value of \"50m\" is greater than the -init-container-cpu-limit value of \"25m\"",
+		},
+		{
+			flags:  []string{"-consul-k8s-image", "foo", "-lifecycle-sidecar-cpu-limit=unparseable"},
+			expErr: "-lifecycle-sidecar-cpu-limit is invalid",
+		},
+		{
+			flags:  []string{"-consul-k8s-image", "foo", "-lifecycle-sidecar-cpu-request=unparseable"},
+			expErr: "-lifecycle-sidecar-cpu-request is invalid",
+		},
+		{
+			flags:  []string{"-consul-k8s-image", "foo", "-lifecycle-sidecar-memory-limit=unparseable"},
+			expErr: "-lifecycle-sidecar-memory-limit is invalid",
+		},
+		{
+			flags:  []string{"-consul-k8s-image", "foo", "-lifecycle-sidecar-memory-request=unparseable"},
+			expErr: "-lifecycle-sidecar-memory-request is invalid",
+		},
+		{
+			flags: []string{"-consul-k8s-image", "foo",
+				"-lifecycle-sidecar-memory-request=50Mi",
+				"-lifecycle-sidecar-memory-limit=25Mi",
+			},
+			expErr: "request must be <= limit: -lifecycle-sidecar-memory-request value of \"50Mi\" is greater than the -lifecycle-sidecar-memory-limit value of \"25Mi\"",
+		},
+		{
+			flags: []string{"-consul-k8s-image", "foo",
+				"-lifecycle-sidecar-cpu-request=50m",
+				"-lifecycle-sidecar-cpu-limit=25m",
+			},
+			expErr: "request must be <= limit: -lifecycle-sidecar-cpu-request value of \"50m\" is greater than the -lifecycle-sidecar-cpu-limit value of \"25m\"",
+		},
 	}
```

Comment:
should we also add a test to check the defaults? Normally, we don't test default flags values, but I feel like in this case these values are very specific, so it warrants a test. What do you think?

Reply:
Should we also add a test that checks that the init container memory limit is greater than the size of the consul binary installed on your path? This might be better to do in the enterprise test since the enterprise binary is bigger than the OSS.

Reply:
I took a look at adding a default test and didn't come across a straightforward approach to the problem. If we want to fully exercise the flags to get at the defaults, it needs to run at the command level. The `Run` command, however, only outputs a return value and/or error which isn't the information we need to verify these defaults. I'm open to suggestions if I've missed something.

Reply:
A test checking the size of the consul binary would be nondeterministic, which is an undesirable trait in a unit test. The tests are set up to run against the consul binary that's currently in the path, which will vary from developer to developer and over time. Adding this sort of thing to the pipeline could make sense, but doesn't belong as a unit test.

Reply:
^ Agreed, adding to this the consul binary varies in size by oss/ent and also by runtime architecture quite dramatically, some are <100M some nearly 130M, so the test may run and pass in our CI but fail in an e.g. darwin64 environment.

Reply:
Yeah don't see how we can test the defaults here.

I think we should (in a separate PR) add a circle ci job that checks the size based on the current image. Not sure if that makes more sense in consul-k8s or consul-helm.

Reply:
About testing defaults, we can do something like this:

```go
func TestFlagDefaults(t *testing.T) {
	cmd := Command{}
	cmd.init()

	require.Equal(t, cmd.flagInitContainerCPULimit, "50m")
}
```

I've run it locally, and it works. 

> A test checking the size of the consul binary would be nondeterministic, which is an undesirable trait in a unit test. 

Yeah, I agree. Running it in CI seems more appropriate.

Reply:
Added default test.